pkg/volume/csi/csi_mounter.go

```diff
@@ -277,17 +278,30 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 		klog.V(2).Info(log("error checking for SELinux support: %s", err))
 	}
 
-	// apply volume ownership
-	// The following logic is derived from https://github.com/kubernetes/kubernetes/issues/66323
-	// if fstype is "", then skip fsgroup (could be indication of non-block filesystem)
-	// if fstype is provided and pv.AccessMode == ReadWriteOnly, then apply fsgroup
-	err = c.applyFSGroup(fsType, mounterArgs.FsGroup, mounterArgs.FSGroupChangePolicy)
-	if err != nil {
-		// At this point mount operation is successful:
-		//   1. Since volume can not be used by the pod because of invalid permissions, we must return error
-		//   2. Since mount is successful, we must record volume as mounted in uncertain state, so it can be
-		//      cleaned up.
-		return volumetypes.NewUncertainProgressError(fmt.Sprintf("applyFSGroup failed for vol %s: %v", c.volumeID, err))
+	fsGroupFeatureGateEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CSIVolumeFSGroupPolicy)
+	// If the feature gate isn't enabled, then adjust the CSIDriver to use the Heuristic
```

Comment:
minor nit: we aren't using `heuristic` anywhere

Reply:
Good catch. This has been removed.